src/main/java/mate/academy/AsyncRequestProcessor.java

```diff
@@ -11,6 +13,19 @@ public AsyncRequestProcessor(Executor executor) {
     }
 
     public CompletableFuture<UserData> processRequest(String userId) {
-        return null;
+        Map<String, UserData> userDataMap = new ConcurrentHashMap<>();
+
+        if (userDataMap.containsKey(userId)) {
+            try {
+                Thread.sleep(500);
+            } catch (InterruptedException e) {
+                throw new RuntimeException();
+            }
+            return CompletableFuture.supplyAsync(() -> userDataMap.get(userId));
+        } else {
+            userDataMap.put(userId, new UserData(userId, "info about user with id: " + userId));
```

Comment:
The `put` operation is not being followed by a `return` statement or any other logic, which means this branch of the conditional does not return a `CompletableFuture`. This is likely a logic error since the method should return a `CompletableFuture` in all execution paths.